### ISSUE TITLE
MODX 3 modUser class joinGroup() crashes is $_SESSION is not set

### ISSUE THREAD

**Issue by BobRay** [modxcms/revolution#15374]
### Summary
Quick summary what's this issue about.

Calling $user->joinGroup() with no $_SESSION set crashes.

Here's the offending code (line 716 of core/model/modx/moduser.class.php):

```
                     unset($_SESSION["modx.user.{$this->get('id')}.userGroupNames"],
                     $_SESSION["modx.user.{$this->get('id')}.userGroups"]);
```

This breaks acceptance tests that run in the MODX Manager.

### Environment
MODX version, MODX 2.8.1

**Comment by BobRay** [comment]
I would propose the fix of adding this line just above the unset call:
```
$_SESSION = isset($_SESSION)? $_SESSION : array();
```

**Comment by sergant210** [comment]
`$_SESSION["modx.user.{$this->get('id')}.userGroupNames"` - this code is legacy. It must be removed. All permission logic must be removed from the session.

**Comment by Ibochkarev** [comment]
Fixed #15376
